File: tests/fixtures/_highlight-corpus/sample.zig
// Zig sample
const std = @import("std");

const Counter = struct {
    value: usize = 0,

    pub fn increment(self: *Counter, n: usize) usize {
        self.value += n;
        return self.value;
    }
};

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    defer _ = gpa.deinit();
    const allocator = gpa.allocator();

    const path = "input.txt";
    const file = try std.fs.cwd().openFile(path, .{});
    defer file.close();

    var buf_reader = std.io.bufferedReader(file.reader());
    var reader = buf_reader.reader();

    var counter = Counter{};
    var line_buf = std.ArrayList(u8).init(allocator);
    defer line_buf.deinit();

    while (true) {
        line_buf.clearRetainingCapacity();
        reader.streamUntilDelimiter(line_buf.writer(), '\n', null) catch |err| switch (err) {
            error.EndOfStream => break,
            else => return err,
        };
        if (line_buf.items.len > 0) {
            _ = counter.increment(line_buf.items.len);
        }
    }

    const stdout = std.io.getStdOut().writer();
    try stdout.print("total bytes: {d}\n", .{counter.value});
}

File: tests/fixtures/_highlight-corpus/sample.c
/* C sample */
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    int value;
} Counter;

static int counter_increment(Counter *c, int n) {
    c->value += n;
    return c->value;
}

int main(void) {
    FILE *f = fopen("input.txt", "r");
    if (f == NULL) {
        perror("fopen");
        return 1;
    }
    Counter c = {0};
    char buf[4096];
    while (fgets(buf, sizeof(buf), f) != NULL) {
        size_t n = strlen(buf);
        if (n > 0 && buf[n - 1] == '\n') {
            n--;
        }
        if (n > 0) {
            counter_increment(&c, (int)n);
        }
    }
    fclose(f);
    printf("total bytes: %d\n", c.value);
    return 0;
}
